lib/order.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'nandos_menu'
+
+class Order
+  def initialize(menu: NandosMenu.new)
+    @order = []
+    @menu = menu
+  end
+
+  def current
+    "Your current order:\n" + order
+  end
+
+  def add(dish, quantity)
+    raise("Item not found on menu") unless on_menu?(dish)
+    @order << { dish => quantity }
+  end
+
+  def total
+    @order.map { |item| price_of_item(item) }.flatten.reduce(:+)
```

Comment:
This logic is a little abbreviated. Could it be made more clear to a reader what's going on here? Maybe using some private methods or variable names?